### ISSUE TITLE
False positives for single line declaration list within multi-line declaration block in declaration-block-semicolon-newline-after

### ISSUE THREAD

**Issue by dcwarwick** [stylelint/stylelint#3153]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

This is a feature request, although I'm not yet sure what to suggest by way of new rule/option/etc so this is also to open a discussion/ask for help/thoughts.

For CSS/SCSS/etc that appears inside style tags I want to require declarations within declaration blocks to always appear on separate lines, and I am therefore using `"declaration-block-semicolon-newline-after": "always"` in my Stylelint config. However, when CSS declarations are used in HTML `style` attributes I do *not* want to require those to appear on separate lines: on the rare occasions when I use inline style attributes I want to be able to do something like `<div style="foo: 1; bar: 2;">`.

> Which rule, if any, is this issue related to?

`declaration-block-semicolon-newline-after`

> What CSS is needed to reproduce this issue?

I would like the following patterns to be considered violations:

Pattern A1:
```css
.foo {
  color: pink; top: 0; width: 500px;
}
```

Pattern A2:
```css
.foo {
  color: pink; top: 0;
  width: 500px;
}
```

Pattern A3:
```html
<div style="
  color: pink; top: 0;
  width: 500px;
">
```

I would like the following patterns _not_ to be considered violations:

Pattern B1:
```css
.foo {
  color: pink;
  top: 0;
  width: 500px;
}
```

Pattern B2:
```html
<div style="color: pink; top: 0; width: 500px;">
```

Pattern B3:
```html
<div style="
  color: pink;
  top: 0;
  width: 500px;
">
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "declaration-block-semicolon-newline-after": "always"
   }
}
```

> Which version of stylelint are you using?

`8.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Via stylelint-webpack-plugin.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

I would like to be able to configure Stylelint so that patterns A1, A2 and A3 are considered *violations* and patterns B1, B2 and B3 are *not* considered violations. This is not currently possible.

> What actually happened (e.g. what warnings or errors you are getting)?

With `"declaration-block-semicolon-newline-after": "always"` patterns A1, A2 and A3 are indeed considered violations, but so is pattern B2.

With `"declaration-block-semicolon-newline-after": "always-multi-line"` patterns B1, B2 and B3 are not considered violations, but neither is pattern A1.

It's as if I would like to set `"declaration-block-semicolon-newline-after"` to be `"always-in-blocks-but-always-multi-line-in-style-attributes`. That isn't my proposal for a property value! I'm not sure what the neatest way to provide for this would be.

**Comment by jeddy3** [comment]
@dcwarwick Thanks for the request and for using the template.

I don't think we ever considered that stylelint would be used to lint the value of a `style` attribute. I didn't even know it could be done! Having said that, it clearly is a valid use-case. 

> With "declaration-block-semicolon-newline-after": "always-multi-line" patterns B1, B2 and B3 are not considered violations, but neither is pattern A1.

I was unable to reproduce this behaviour. 

Given:

```css
.foo {
  color: pink; top: 0; width: 500px;
}
```

Using the demo tool, `"declaration-colon-newline-after": "always-multi-line",` produces two violations:

```shell
2:15 error Expected newline after ";" in a multi-line declaration block (declaration-block-semicolon-newline-after)
2:23 error Expected newline after ";" in a multi-line declaration block (declaration-block-semicolon-newline-after)
```

Are you able to create a reproducible repo?

Hopefully, there is just a bug here and our `*-multi-line` options work out well here (which would be quite fortuitous as we hadn't considered with use-case when coming up with those options) i.e. the following config:

```json
{
  "declaration-colon-newline-after": "always-multi-line",
  "declaration-colon-space-after": "always-single-line",
  "declaration-colon-space-before": "never"
}
```

Allows your "B" patterns whilst producing violations for your "A" patterns.

**Comment by jeddy3** [comment]
Closing for now. Feel free to reopen with a reproducible repo and/or clarification.
